apps/api/src/app/tenant/usecases/delete-tenant/delete-tenant.usecase.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+
+import { TenantRepository, DalException } from '@novu/dal';
+
+import { DeleteTenantCommand } from './delete-tenant.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+
+@Injectable()
+export class DeleteTenant {
+  constructor(private tenantRepository: TenantRepository) {}
+
+  async execute(command: DeleteTenantCommand) {
+    const tenant = await this.tenantRepository.findOne({
+      environmentId: command.environmentId,
+      organizationId: command.organizationId,
+      identifier: command.identifier,
+    });
+
+    if (!tenant) {
+      throw new NotFoundException(
+        `Tenant with identifier: ${command.identifier} is not exists under environment ${command.environmentId}`
+      );
+    }
```

Comment:
Reuse the `get-tenant` use case to avoid to have to reimplement this.

Reply:
good idea!